lib/did_you_mean/experimental/key_error_name_correction.rb

```diff
@@ -1,31 +1,43 @@
 module DidYouMean
   module Experimental
-    module KeyErrorWithNameAndKeys
-      FILE_REGEXP = %r"#{Regexp.quote(__FILE__)}"
+    if KeyError.method_defined?(:receiver) && KeyError.method_defined?(:key)
```

Comment:
You don't have to worry about backward-compatibility since GH master is only compatible with the latest Ruby (`2.5.0-dev` at the time of writing).